src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/CreateMergedManifestBuildModelTests.cs

```diff
@@ -0,0 +1,317 @@
+            };
+
+            expectedBuildModel.Artifacts =
+                new ArtifactSet
+                {
+                    Packages = new List<PackageArtifactModel> { shippingPackageArtifact },
+                    Blobs = new List<BlobArtifactModel> { blobArtifactModel }
+                };
+
+            BuildModel actualModel = pushMetadata.CreateMergedManifestBuildModel(
+                ImmutableList.Create(dataInBlobSet, shippingAssetData), manifestBuildData);
+
+            actualModel.Should().BeEquivalentTo(expectedBuildModel);
+        }
+
+        [Test]
+        public void GivenNonShippingAssets()
+        {
+            BuildModel expectedBuildModel = GetBuildModel();
+            PushMetadataToBuildAssetRegistry pushMetadata = GetPushMetadata();
+
+            PackageArtifactModel packageArtifact = new PackageArtifactModel
+            {
+                Attributes = new Dictionary<string, string>
+                    {
+                        { "NonShipping", "true" },
+                    },
+                Id = nonShippingAssetData.Name,
+                Version = nonShippingAssetData.Version
+            };
+
+            expectedBuildModel.Artifacts =
+                new ArtifactSet
+                {
+                    Packages = new List<PackageArtifactModel> { packageArtifact }
+                };
+
+            BuildModel actualModel = pushMetadata.CreateMergedManifestBuildModel(
+                ImmutableList.Create(nonShippingAssetData), manifestBuildData);
+
+            actualModel.Should().BeEquivalentTo(expectedBuildModel);
+        }
+
+        [Test]
+        public void GivenShippingAssets()
+        {
+            BuildModel expectedBuildModel = GetBuildModel();
+            PushMetadataToBuildAssetRegistry pushMetadata = GetPushMetadata();
+
+            PackageArtifactModel packageArtifact = new PackageArtifactModel
+            {
+                Attributes = new Dictionary<string, string>
+                    {
+                        { "NonShipping", "false" },
+                    },
+                Id = shippingAssetData.Name,
+                Version = shippingAssetData.Version
+            };
+
+            expectedBuildModel.Artifacts =
+                new ArtifactSet
+                {
+                    Packages = new List<PackageArtifactModel> { packageArtifact }
+                };
+
+            BuildModel actualModel = pushMetadata.CreateMergedManifestBuildModel(
+                ImmutableList.Create(shippingAssetData), manifestBuildData);
+
+            actualModel.Should().BeEquivalentTo(expectedBuildModel);
+        }
+
+        [Test]
+        public void GivenNullAssetsList_ExpectNullReferenceException()
+        {
+            PushMetadataToBuildAssetRegistry pushMetadata = GetPushMetadata();
+
+            Action act = () => pushMetadata.CreateMergedManifestBuildModel(null, manifestBuildData);
+            act.Should().Throw<NullReferenceException>();
```

Comment:
It's odd to assert "NullReferenceException". That usually means incorrect behavior because the null wasn't checked early enough in the code, and weird side effects happen.

I'd either remove the test or fix the code to throw a more useful exceptoin (ArgumentNullException would be correct in this case).

This test is essentially forcing incorrect behavior.

Reply:
I'll pull it for the moment since I don't think that there are null checks anywhere in this class and that's a bigger project than I'm budgeted for at the moment. I'll put in an issue for it so that it's at least documented and can be put on the bottom of the pile by people who make the decisions. :)